src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,149 @@
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { PeopleTable } from '../components/PeopleTable';
+import { PeopleFilters } from '../components/PeopleFilters';
+import { Loader } from '../components/Loader';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  const fetch = async () => {
+    setIsLoading(true);
+
+    try {
+      const peopleFromServer = await getPeople();
+
+      setPeople(peopleFromServer);
+    } catch {
+      setIsError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const [urlParams] = useSearchParams();
+
+  const query = urlParams.get('query');
+  const centuries = urlParams.getAll('centuries');
+  const sortAscParam = urlParams.get('sort') || '';
+  const sortDescParameter = urlParams.get('order');
+
+  const compareQueryMatch = (names: (string | null)[]) => {
+    if (query) {
+      return (
+        names.some(name => {
+          if (name) {
+            return name.toLowerCase().includes(query.toLowerCase());
+          }
+
+          return false;
+        })
+      );
+    }
+
+    return false;
+  };
+
+  const filteredPeople = people.filter(person => {
+    const { name, motherName, fatherName } = person;
+    const personCentury = (+person.born.toString().slice(0, 2) + 1).toString();
+
+    let isQueryMatch;
+    let isCenturyMatch;
+
+    if (!query || compareQueryMatch([name, motherName, fatherName])) {
+      isQueryMatch = true;
+    }
+
+    if (!centuries.length || centuries.includes(personCentury)) {
+      isCenturyMatch = true;
+    }
+
+    const matchAll = isQueryMatch && isCenturyMatch;
+
+    switch (urlParams.get('sex')) {
+      case null: return matchAll;
+      case 'm': return (person.sex === 'm' && matchAll);
+      case 'f': return (person.sex === 'f' && matchAll);
+
+      default: return person.sex === urlParams.get('sex') && matchAll;
+    }
```

Comment:
```suggestion
    return urlParams.get('sex') 
      ? person.sex === urlParams.get('sex') && matchAll 
      : matchAll;
```

Reply:
switch (urlParams.get('sex')) {
  case 'm': return (person.sex === 'm' && matchAll);
  case 'f': return (person.sex === 'f' && matchAll);
}

is the same with 
return person.sex === urlParams.get('sex') && matchAll

Reply:
Now I got it